File: android/app/src/main/java/embrun/fr/tournoibridgeonline/common/model/MouvementResult.kt

    package embrun.fr.tournoibridgeonline.common.model
/*    sealed class MouvementResult {
        data class Complet(val mouvement: Mouvement, val tousTermines: Boolean = true) : MouvementResult()
        object Terminé : MouvementResult()
        data class ClassementEnAttente(val nbreEnregistrement: Int) : MouvementResult()
        data class Erreur(val message: String) : MouvementResult()
    }*/
    sealed class MouvementResult {
        data class Complet(val mouvement: Mouvement, val tousTermines: Boolean = true) : MouvementResult()
        data class ClassementEnAttente(val nbreEnregistrement: Int) : MouvementResult()
        data class Erreur(val message: String) : MouvementResult()
    }

File: web/asso avec mitchell/MouvementResult.php
<?php  


class MouvementResult {
    public string $status = "";
    public string $message = "";
    public ?Mouvement $mouvement = null;
    public ?int $nbreEnregistrement = null;
    public bool $tousTermines = true;

    public static function MouvementComplet(Mouvement $mouvement, bool $tousTermines = true): self {
        $result = new self();
        $result->status      = "MouvementComplet";
        $result->mouvement   = $mouvement;
        $result->tousTermines = $tousTermines;
        return $result;
    }

    public static function ClassementEnAttente(int $nbreEnregistrement): self {
        $result = new self();
        $result->status             = "ClassementEnAttente";
        $result->nbreEnregistrement = $nbreEnregistrement;
        return $result;
    }

    public static function Erreur(string $message): self {
        $result = new self();
        $result->status  = "Erreur";
        $result->message = $message;
        return $result;
    }
}